eo-maven-plugin/src/main/java/org/eolang/maven/name/ObNmDefault.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2016-2023 Objectionary.com
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.eolang.maven.name;
+
+import org.cactoos.Scalar;
+import org.cactoos.scalar.Sticky;
+import org.cactoos.scalar.Unchecked;
+import org.eolang.maven.hash.CommitHash;
+
+/**
+ * Object name with default hash.
+ *
+ * @since 0.29.6
+ */
+public final class ObNmDefault implements ObjectName {
+
+    /**
+     * Full name split.
+     */
+    private final Unchecked<String[]> split;
+
+    /**
+     * Ctor.
+     *
+     * @param object Object full name (with version or not).
+     * @param def Default hash if version in full name is absent.
+     */
+    public ObNmDefault(final String object, final CommitHash def) {
+        this.split = new Unchecked<>(
```

Comment:
@maxonfjvipon We don't need `Unchecked` and `Sticky` objects here, because we have caching in another class.
Moreover, it's better to make constructors `code-free`:
https://www.yegor256.com/2015/05/07/ctors-must-be-code-free.html
Hence, I believe we have to use `divided` method in methods, not in a constructor. (Even if it is less effective.)